lib/sw-precache.js

```diff
@@ -47,9 +51,13 @@ function getFileAndSizeAndHashForFile(file) {
   return null;
 }
 
-function getFilesAndSizesAndHashesForGlobPattern(globPattern) {
+function getFilesAndSizesAndHashesForGlobPattern(globPattern, excludeFilePath) {
   return glob.sync(globPattern.replace(path.sep, '/')).map(function(file) {
-    return getFileAndSizeAndHashForFile(file);
+    // Return null if we want to exclude this file, and it will be excluded in
+    // the subsequent filter().
+    return excludeFilePath === absolutePath(file) ?
+      null :
```

Comment:
Koolio - ta Jeff.

On Thu, 21 Apr 2016, 15:51 Jeffrey Posnick, <notifications@github.com>
wrote:

> In lib/sw-precache.js
> <https://github.com/GoogleChrome/sw-precache/pull/107#discussion_r60594149>
> :
>
> >    return glob.sync(globPattern.replace(path.sep, '/')).map(function(file) {
> > -    return getFileAndSizeAndHashForFile(file);
> > +    // Return null if we want to exclude this file, and it will be excluded in
> > +    // the subsequent filter().
> > +    return excludeFilePath === absolutePath(file) ?
> > +      null :
>
> You save an unneeded fs.statSync() by filtering it out early and avoiding
> the call to getFileAndSizeAndHashForFile(). That was the main reason.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/GoogleChrome/sw-precache/pull/107/files/be0bd63f40fea5dcd5efa1cf799fb47d43afebe6#r60594149>
>